File: KfzScala/src/kfzwerkstatt/Werkstatt.scala

package kfzwerkstatt

class Werkstatt {
  var mitarbeiter: List[Any] = List[Any]()
  var kunden: List[Kunde] = List[Kunde]()


  def addMitarbeiterVerkaeufer(v: Verkaeufer) {
    mitarbeiter = v :: mitarbeiter 
  }

  def addMitarbeiterMechaniker(m: Mechaniker) {
    mitarbeiter = m :: mitarbeiter 
  }

  def addKunde(k: Kunde) {
    kunden = k :: kunden
  }

  def gibNettoEinnahmenAus {
    var summe: Int = 0
    import scala.collection.JavaConversions._
    for (o <- mitarbeiter) {
      if (o.isInstanceOf[Verkaeufer]) {
        summe += (((o.asInstanceOf[Verkaeufer]).sumVerkaeufe) * 0.81).asInstanceOf[Int]
      }
      else if (o.isInstanceOf[Mechaniker]) {
        summe += (((o.asInstanceOf[Mechaniker]).gibArbeitskostenZureuck) * 0.81).asInstanceOf[Int]
      }
    }
    System.out.println("Netto :" + summe)
  }

  def gibBruttoEinnahmenAus {
    var summe: Int = 0
    import scala.collection.JavaConversions._
    for (o <- mitarbeiter) {
      if (o.isInstanceOf[Verkaeufer]) {
        summe += ((o.asInstanceOf[Verkaeufer]).sumVerkaeufe).asInstanceOf[Int]
      }
      else if (o.isInstanceOf[Mechaniker]) {
        summe += ((o.asInstanceOf[Mechaniker]).gibArbeitskostenZureuck)
      }
    }
    System.out.println("Brutto :" + summe)
  }

  def gibKundenbetraegeAus {
    System.out.println("Kundenbetraege ")
    import scala.collection.JavaConversions._
    for (o <- kunden) {
      val k: Kunde = o.asInstanceOf[Kunde]
      System.out.println(k.vorname + " " + k.name + " : " + k.bezahlt)
    }
  }
}

File: Kfz/src/kfzwerkstatt/Werkstatt.java
package kfzwerkstatt;

import java.util.ArrayList;
import java.util.List;

public class Werkstatt {

	private List mitarbeiter;
	private List kunden;

	public Werkstatt() {
		mitarbeiter = new ArrayList();
		kunden = new ArrayList();
	}

	public void addMitarbeiterVerkaeufer(Verkaeufer v) {
		mitarbeiter.add(v);
	}

	public void addMitarbeiterMechaniker(Mechaniker m) {
		mitarbeiter.add(m);
	}

	public void addKunde(Kunde k) {
		kunden.add(k);
	}

	public void gibNettoEinnahmenAus() {
		int summe = 0;

		for (Object o : mitarbeiter) {
			if (o instanceof Verkaeufer) {
				summe += (((Verkaeufer) o).sumVerkaeufe) * 0.81; // 19% Mwst
				// abgezogen
			} else if (o instanceof Mechaniker) {
				summe += (((Mechaniker) o).gibArbeitskostenZureuck()) * 0.81;
			}
		}// for

		System.out.println("Netto :" + summe);
	}

	public void gibBruttoEinnahmenAus() {
		int summe = 0;

		for (Object o : mitarbeiter) {
			if (o instanceof Verkaeufer) {
				summe += (((Verkaeufer) o).sumVerkaeufe);
			} else if (o instanceof Mechaniker) {
				summe += (((Mechaniker) o).gibArbeitskostenZureuck());
			}
		}// for
		System.out.println("Brutto :" + summe);
	}

	public void gibKundenbetraegeAus() {
		System.out.println("Kundenbetraege ");
		for (Object o : kunden) {
			Kunde k = (Kunde) o;
			System.out.println(k.getVorname() + " " + k.getName() + " : " + k.getBezahlt()); 
		}
	}
}
